pkg/util/fsfreeze.go

```diff
@@ -23,14 +28,25 @@ const (
 	unfreezeTimeout = 5 * time.Second
 )
 
-// GetFreezePointFromEndpoint returns the absolute path to the canonical location we will try to mount a filesystem to
-// before freezing it.
-func GetFreezePointFromEndpoint(endpoint string) string {
-	return filepath.Join(freezePointDirectory, filepath.Base(endpoint))
+// GetFreezePointFromDevicePath returns the absolute path to the canonical location we will try to mount a filesystem to
+// before freezeing it.
+func GetFreezePointFromDevicePath(devicePath string) string {
+	return GetFreezePointFromVolumeName(filepath.Base(devicePath))
 }
 
-// AttemptFreezeFilesystem attempts to freeze the filesystem mounted at freezePoint.
-func AttemptFreezeFilesystem(freezePoint string, exec lhexec.ExecuteInterface) error {
+// GetFreezePointFromVolumeName returns the absolute path to the canonical location we will try to mount a filesystem to
+// before freezeing it.
+func GetFreezePointFromVolumeName(volumeName string) string {
+	return filepath.Join(freezePointDirectory, volumeName)
+}
+
+// GetDevicePathFromVolumeName mirrors longhorndev.getDev. It returns the device path that go-iscsi-helper will use.
+func GetDevicePathFromVolumeName(volumeName string) string {
+	return filepath.Join(longhorndev.DevPath, volumeName)
+}
```

Comment:
Should these functions return a non-empty value (`/var/lib/longhorn/freeze/`) when the input is empty?

Reply:
If I understand you correctly, you are saying it is safer if we return an empty value instead of whatever the base of the join is? Makes sense to me.